### PR TITLE
[DF-375] [DF-524] GTM tag for when a user select a filter

### PR DESCRIPTION
**Pull request by janehughes**
Ticket URLs:
https://national-archives.atlassian.net/browse/DF-375 (Tracking when a user successfully selects a checkbox filter)
https://national-archives.atlassian.net/browse/DF-524 (Tracking when a user selects a date filter)

## About these changes

Adds JS to push active search filters separately to the datalayer for GTM.

## How to check these changes

Apply some filters to an active search, open the console and type in 'dataLayer' to check that the active filters are being sent  to the datalayer correctly. 

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [  ] Added/updated tests and documentation where relevant. N/A

## Merging a feature or bug fix?

1. Use the `Squash and merge` option to keep the target branch's commit history nice and clean.
2. Where relevant, include the ticket number in commit title, e.g. `DF-XXX: Ticket name / short description`.

## Merging a release branch?

1. Use the `Create a merge commit` option to preserve the original commit IDs.
2. Use the message format `release/<major.minor.patch>` when merging.

## Deployment guidance

1. Post to the appropriate Slack channel to check that it is okay to continue.
2. To start a deployment, push the relevant branch to the `platform` remote.
3. Update the appropriate Slack channel when the deployment is complete.
4. Update the status on the corresponding Jira ticket (where relevant).

### CODE REVIEW

**Review by JohnHeeryTNA — COMMENTED** [review]
Hi @janehughes : the **js** looks good to me but a few questions on `templates/search/blocks/search_results__selected-filters.html`

We have added in custom attributes to the list item (line 9) but I think these attributes should actually be added to the href item (line 11). Please check with @hoake. 

We also have an attribute in there which doesn't seem to get populated? (line 9)
`data-filter`, is that correct?

File: templates/search/blocks/search_results__selected-filters.html [hunk 1]
```diff
@@ -6,7 +6,7 @@ <h2 class="sr-only">Selected filters</h2>
     {% for field_name, selected_options in selected_filters.items %}
         {% for value, label in selected_options %}
             {# IMPORTANT: Never apply '|safe' to value or label here, as validation is not applied to most of these fields #}
-            <li class="search-results__selected-filters-item">
+            <li class="search-results__selected-filters-item" data-filter data-filter-value="{{ field_name }}" data-filter-name="{{ label }}" data-search-bucket="{{ buckets.current.label }}">
```
Comment (TNA-Allan) [inline templates/search/blocks/search_results__selected-filters.html:9]:
```suggestion
            <li class="search-results__selected-filters-item" data-filter data-filter-value="{{ label }}" data-filter-name="{{ field_name }}" data-search-bucket="{{ buckets.current.label }}">
```

Comment (TNA-Allan) [reply templates/search/blocks/search_results__selected-filters.html:9]: For date filter, you might want to check `data-filter-value=` if {{label}} or {{value}} is Ok.

Comment (janehughes) [reply templates/search/blocks/search_results__selected-filters.html]: Thanks @TNA-Allan you're right I had these the wrong way round - updated now.

**Comment by janehughes** [comment]
> Hi @janehughes : the **js** looks good to me but a few questions on `templates/search/blocks/search_results__selected-filters.html`
> 
> We have added in custom attributes to the list item (line 9) but I think these attributes should actually be added to the href item (line 11). Please check with @hoake.
> 
> We also have an attribute in there which doesn't seem to get populated? (line 9) `data-filter`, is that correct?

Thanks @JohnHeeryTNA - I don't think it matters in this case as we're not tracking link interactions, we just need to grab the values and send them to the dataLayer - but I will clarify with @hoake 👍 

re: the data-filters attribute - this is only used for the JS to grab the active filters, it isn't used for tracking. Let me know if it'd be helpful for me to run through this.

File: templates/search/blocks/search_results__selected-filters.html (hunk 1 above)
Comment (TNA-Allan) [reply templates/search/blocks/search_results__selected-filters.html]: Hi @janehughes , as discussed for date fields the values the format is `YYYY-MM-DD` as per [DF-524-Tracking when a user selects a date filter](https://national-archives.atlassian.net/browse/DF-524)

Comment (janehughes) [reply templates/search/blocks/search_results__selected-filters.html]: @TNA-Allan managed to fix whatever was going on with my local build - have made and tested the changes for the date fields now so it's ready for a final review now :) thanks

File: templates/search/blocks/search_results__selected-filters.html [hunk 2]
```diff
@@ -6,7 +6,7 @@ <h2 class="sr-only">Selected filters</h2>
     {% for field_name, selected_options in selected_filters.items %}
         {% for value, label in selected_options %}
             {# IMPORTANT: Never apply '|safe' to value or label here, as validation is not applied to most of these fields #}
-            <li class="search-results__selected-filters-item">
+            <li class="search-results__selected-filters-item" data-filter data-filter-value="{% if field_name == 'opening_start_date' %}{{ value|date:"o-m-d" }}{% elif field_name == 'opening_end_date' %}{{ value|date:"o-m-d" }}{% else %}{{ label }}{% endif %}" data-filter-name="{{ field_name }}" data-search-bucket="{{ buckets.current.label }}">
```
Comment (TNA-Allan) [inline templates/search/blocks/search_results__selected-filters.html:9]: Thanks @janehughes . It looks good. If we have more date filter fields, it looks like a candidate for refactor.

**Review by TNA-Allan — APPROVED** [review]
Thanks @janehughes. It looks good. From FE side, please can @JohnHeeryTNA review and approve and will then merge.

File: templates/search/blocks/search_results__selected-filters.html (hunk 2 above)
Comment (janehughes) [reply templates/search/blocks/search_results__selected-filters.html:9]: Yep, agreed.

**Review by JohnHeeryTNA — APPROVED** [review]
approving change

**Pull request merged**